File: python/all-your-base/all_your_base.py
from functools import reduce


def rebase(input_base, digits, output_base):
    if input_base < 2:
        raise ValueError("input base must be >= 2")
    if output_base < 2:
        raise ValueError("output base must be >= 2")
    if any(d >= input_base for d in digits):
        raise ValueError("all digits must satisfy 0 <= d < input base")
    if any(d < 0 for d in digits):
        raise ValueError("all digits must satisfy 0 <= d < input base")
    int_sum = reduce(lambda acc, n: acc * input_base + n, digits, 0)
    if int_sum == 0:
        return [0]
    new_digits = []
    while int_sum > 0:
        new_digits.insert(0, int_sum % output_base)
        int_sum = int_sum // output_base
    return new_digits

File: r/all-your-base/all-your-base.R
rebase = function(from_base, digits, to_base) {
  if (from_base < 2) stop("input base must be >= 2")
  if (to_base < 2) stop("output base must be >= 2")
  if (any(digits < 0 | digits >= from_base)) stop("all digits must satisfy 0 <= d < input base")
  num = 0
  if (length(digits) < 1) return (0)
  for (i in 1:length(digits)) {
    num = num * from_base + digits[[i]]
  }
  output <- c()
  if (num == 0) return(0)
  while (num > 0) {
    output <- c(num %% to_base, output)
    num <- num %/% to_base
  }
  output
}
